pages/api/createPlanApi.js

```diff
@@ -0,0 +1,19 @@
+import createPlan from '../../lib/dependencies';
+
+export default async (req, res) => {
+  if (req.method === 'POST') {
+    try {
+      const result = await createPlan.execute({
+        firstName: req.body.firstName,
+        lastName: req.body.lastName
+      });
+
+      res.status(200).json(result);
+    } catch (err) {
+      //log error here
+      res.status(500).json({
+        error: `could not create a plan with first name: ${req.body.firstName}, last name: ${req.body.lastName}`
+      });
```

Comment:
Maybe worth differentiating between validation problems (e.g. we throw an error if `lastName` is null) and then returning a 400 instead? Could be useful on the front-end to determine what sort of error message to show?

Reply:
yeah that sounds like a good idea